### PR TITLE
Implement bayesian filter

### PR DESCRIPTION
**Pull request by robgineer**

### CODE REVIEW

File: bayesian_filter.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    }
+    return belief;
+  }
+} // unnamed namespace
+
+void BayesianFilter::generateBelief(const SensorData& sensor_data) {
+  
+  belief.sensor_1_x_belief = getBelief(sensor_data.sensor_1_x_confidence);
+  belief.sensor_1_y_belief = getBelief(sensor_data.sensor_1_y_confidence);
+}
```
Comment (yannlabou) [inline bayesian_filter.cpp:37]: Shouldn't `generateBelief` returns a belief?

File: bayesian_filter.h [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+{
+  std::vector<double> sensor_1_x_belief;
+  std::vector<double> sensor_1_y_belief;
+};
+
+class BayesianFilter
+{
+public:
+  // sensor data incl. ground truth
+  Belief belief;
```
Comment (yannlabou) [inline bayesian_filter.h:18]: Should be a `private` member.

File: bayesian_filter.cpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
+#include "bayesian_filter.h"
+#include "sensor_data_handler.h"
+#include <iostream>
```
Comment (yannlabou) [inline bayesian_filter.cpp:3]: Probably not needed here.

File: sensor_data_handler.cpp [hunk 4]
```diff
@@ -36,4 +36,21 @@ void SensorDataHandler::readSensorData(std::string_view file_path)
...
+
+int SensorDataHandler::writeSensorData(std::string_view filename,
+                                        std::string_view column_name_1, std::vector<double> values_1,
+                                        std::string_view column_name_2, std::vector<double> values_2) {
+    // get filename
+    std::ofstream csv_file(filename.data());
+    // write header
+    csv_file << column_name_1 << ";" << column_name_2 << "\n";
+    // error handling
+    if (values_1.size() != values_2.size()) {return 1;}
```
Comment (yannlabou) [inline sensor_data_handler.cpp:49]: We should check it first. If this fails, the `csv_file` won't be closed.

File: sensor_data_handler.h [hunk 5]
```diff
@@ -23,6 +26,9 @@ class SensorDataHandler
   // reader
   void readSensorData(std::string_view file_path);
 
-  // TODO
-  void writeSensorData(SensorData sensor_data);
-};
\ No newline at end of file
+  // writer
+  int writeSensorData(std::string_view filename, std::string_view column_name_1, std::vector<double> values_1,
```
Comment (yannlabou) [inline sensor_data_handler.h:30]: I suggest `bool ` as return type.

**Review by yannlabou — CHANGES_REQUESTED** [review]

File: bayesian_filter.cpp (hunk 1 above)
Comment (robgineer) [reply bayesian_filter.cpp:37]: The idea was that it should generate one and store it somewhere...

File: bayesian_filter.h (hunk 2 above)
Comment (robgineer) [reply bayesian_filter.h:18]: Well, in that case we would need the "trivial" getters that we didnt like ;) 
(We are accessing the member within ```run_flter``` directly).

File: bayesian_filter.cpp (hunk 3 above)
Comment (robgineer) [reply bayesian_filter.cpp]: Done. d51e52f237007965ab3fc1c0cba58bc5b1c917f6

File: sensor_data_handler.h (hunk 5 above)
Comment (robgineer) [reply sensor_data_handler.h]: Done. d51e52f237007965ab3fc1c0cba58bc5b1c917f6

**Pull request merged**